src/common/hook/usePublishQueue/utils/publish.tsx

```diff
@@ -1,25 +1,65 @@
-import { FormEntry, Config, RawDocument, PublishingJob } from "../../../../types";
-import { defaultsDeep, groupBy } from "lodash";
 import { wrapDocuments } from "@govtechsg/open-attestation";
+import { defaultsDeep, groupBy } from "lodash";
+import { Config, FormEntry, PublishingJob, RawDocument } from "../../../../types";
+import { getQueueNumber } from "../../../API/storageAPI";
+import { encodeQrCode } from "../../../utils";
 
-export const getRawDocuments = (forms: FormEntry[], config: Config): RawDocument[] => {
-  return forms.map(({ data, templateIndex, fileName, ownership }) => {
-    const formConfig = config.forms[templateIndex];
-    if (!formConfig) throw new Error("Form definition not found");
-    const formDefaults = formConfig.defaults;
-    const formData = { ...data.formData };
-    defaultsDeep(formData, formDefaults);
-    const contractAddress =
-      formData.issuers[0]?.documentStore || formData.issuers[0]?.tokenRegistry;
-    const payload = formConfig.type === "TRANSFERABLE_RECORD" ? { ownership } : {};
-    return {
-      type: formConfig.type,
-      contractAddress,
-      rawDocument: formData,
-      fileName,
-      payload,
-    };
-  });
+interface QueueNumberTypes {
+  id: string;
+  key: string;
+}
+
+interface QrURLObj {
+  links: { self: { href: string } };
+}
+
+const getQrURL = (queueNumber: QueueNumberTypes, network: string): QrURLObj => {
+  const qrCodeObject = { links: { self: { href: "" } } };
+  if (!queueNumber.id) return qrCodeObject;
+
+  const qrURLObj = {
```

Comment:
https://stackoverflow.com/questions/15526107/acronyms-in-camelcase

Reply:
ok